### ISSUE TITLE
How to use this architecture with custom dataset

### ISSUE THREAD

**Issue by siftr** [vonclites/squeezenet#2]
@vonclites Hello, 
Thanks for the architecture in tensorflow. I have trained squeezenet on caffe with my custom dataset and being very new to tensorflow(started it 2 days back) I am facing difficulty in understanding how to use this architecture for training on my custom data. I have trained it with Cifar but now I want to train it with my on custom data.
Any directions on how to train it understanding my non-familarity with the platform might be very helpful.

Thanks!

**Comment by siftr** [comment]
Thanks! I was able to achieve it :)

**Comment by apockill** [comment]
Hi siftr, could you pass me some tips on how you were able to do this?

**Comment by vonclites** [comment]
I'm in the process of updating the program to incorporate the tf.data API.  That should make training on your own TFRecords easier.

**Comment by vonclites** [comment]
OK.  Pushed an updated version.  Seems to be working for me.  Dunno if it will for you.  But it looks prettier!

**Comment by Scaglia** [comment]
Hi, I was able to create 2 TFRecord using my custom dataset (one for training and one for test), but I'm having some problem to train the squeezenet.
I'm currently looking for just 2 classes (labeled 0 and 1), and I'm working on a build machine without GPU, so I already provided to modify the code for my case, but I still have some problems:

- in `_clone_def()` I had to modify the label argument of `tf.losses.sparse_softmax_cross_entropy` with `tf.convert_to_tensor(np.array([[1,0]]))` as stated [here](https://stackoverflow.com/questions/49083984/valueerror-can-not-squeeze-dim1-expected-a-dimension-of-1-got-3-for-sparse) (btw I know it is not correct, I was just trying)
- in `_generate_metrics()` I have a similar error, but I have no idea on how to fix it: `tf.metrics.accuracy(
                    labels=self.labels,
                    predictions=predictions
                )
` report the same error as before.

This is my first attempt in training a neural network, please be gentle :)

P.s. I'm using tensorflow 1.10 and unfortunately I can not upgrade for what I need
